Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -1550,20 +1574,57 @@ private final class TokenStreamCreator: SyntaxVisitor {
     extractLeadingTrivia(token)
     appendBeforeTokens(token)
 
-    let text: String
-    if token.leadingTrivia.hasBackticks && token.trailingTrivia.hasBackticks {
-      text = "`\(token.text)`"
+    if let pendingSegmentIndex = pendingMultilineStringSegmentPrefixLengths.index(forKey: token) {
+      appendMultilineStringSegments(at: pendingSegmentIndex)
     } else {
-      text = token.text
+      // Otherwise, it's just a regular token, so add the text as-is.
+      let text: String
+      if token.leadingTrivia.hasBackticks && token.trailingTrivia.hasBackticks {
+        text = "`\(token.text)`"
+      } else {
+        text = token.text
+      }
+
+      appendToken(.syntax(text))
     }
-    appendToken(.syntax(text))
 
     appendAfterTokensAndTrailingComments(token)
 
     // It doesn't matter what we return here, tokens do not have children.
     return .skipChildren
   }
 
+  /// Appends the contents of the pending multiline string segment at the given index in the
+  /// registration dictionary (removing it from that dictionary) to the token stream, splitting it
+  /// into lines along with `.newline` tokens and stripping the leading whitespace.
+  private func appendMultilineStringSegments(at index: Dictionary<TokenSyntax, Int>.Index) {
+    let (token, prefixCount) = pendingMultilineStringSegmentPrefixLengths[index]
+    pendingMultilineStringSegmentPrefixLengths.remove(at: index)
+
+    let lines = token.text.split(separator: "\n", omittingEmptySubsequences: false)
+
+    // The first "line" is a special case. If it is non-empty, then it is a piece of text that
+    // immediately followed an interpolation segment on the same line of the string, like the
+    // " baz" in "foo bar \(x + y) baz". If that is the case, we need to insert that text before
+    // anything else.
+    let firstLine = lines.first!
+    if !firstLine.isEmpty {
+      appendToken(.syntax(String(firstLine)))
+    }
+
+    // Add the remaining lines of the segment, preceding each with a newline and stripping the
+    // leading whitespace so that the pretty-printer can re-indent the string according to the
+    // standard rules that it would apply.
+    for line in lines.dropFirst() as ArraySlice {
+      appendToken(.newline)
+
+      let remainingString = line.dropFirst(prefixCount)
```

Comment:
Could this be destructive to the string if it was malformed by not having sufficient leading spaces on some line? I'm pretty sure that would **not** compile, so that may not be a worthwhile consideration. It looks like the syntax parser is able to parse such a malformed string into a StringLiteralExpr.

Say if you wrote something like this, and then formatted it (even though it doesn't compile), maybe because you format on save:
```
  let someString = """
my first good string
  """
=>
  let someString = """
   first good string
  """
```

Reply:
Thanks for catching that; this stage of the parser is more forgiving than I was expecting.

I've updated this so that lines that contain fewer than the expected number of spaces only strip off the spaces that are there; this forces those lines to align, which I think is a nice "autocorrection" vs. bailing out entirely and leaving the string untouched.

I also realized there were errors in blank line handling because the use of `.newline` tokens was triggering the folding logic when they exceeded the maximum number allowed by the configuration, so the latest version I pushed also includes some necessary changes to the `.newline` logic to support newlines that should always _always_ **_always_** be there.